src/supplemental/nanolib/mqtt_db.c

```diff
@@ -199,6 +228,12 @@ dbtree_get_tree(dbtree *db, void *(*cb)(uint32_t pipe_id))
 		dbtree_info **ret_line_ping = NULL;
 		for (size_t i = 0; i < cvector_size(nodes); i++) {
 			dbtree_info *vn = nni_zalloc(sizeof(dbtree_info));
+			if (vn == NULL) {
+				/* All vn are pushed in ret_line_ping vector*/
+				cvector_free(ret_line_ping);
```

Comment:
Can you check that using cvector_free to release memory is correct or not ? @JaylinYu

Reply:
seems ok to me
@lee-emqx

Reply:
Have you noticed any issues with cvector_free?

Reply:
I haven't had any issues yet.
Just curious about the rationality of this approach :P

Reply:
When dealing with pointer arrays such as ret_line_ping, it's important to remember to free its members before freeing the array itself (use cvector_free). This ensures that memory is properly deallocated and avoids any potential issues.

Reply:
I just looked at the cvector_free function and did not free all the members, so I added the following code. @JaylinYu @lee-emqx 
```
				for (int j = 0; j < cvector_size(ret_line_ping); j++) {
					nni_free(ret_line_ping[j], sizeof(dbtree_info));
				}
```

Reply:
vn's member also need :P